storage/badger/operation/transaction_results.go

```diff
@@ -69,3 +71,30 @@ func LookupTransactionResultsByBlockIDUsingIndex(blockID flow.Identifier, txResu
 
 	return traverse(makePrefix(codeTransactionResultIndex, blockID), txErrIterFunc)
 }
+
+// RemoveTransactionResultsByBlockID removes the transaction results for the given blockID
+func RemoveTransactionResultsByBlockID(blockID flow.Identifier) func(*badger.Txn) error {
+	return func(txn *badger.Txn) error {
+		var txResults []flow.TransactionResult
+
+		err := SkipNonExist(LookupTransactionResultsByBlockID(blockID, &txResults))(txn)
```

Comment:
Not sure how performance relevant this code is. But I assume in a common case, we are removing thousands maybe even hundred out thousand entries. So it might take a while.

There are some comments for performance [here](https://github.com/dgraph-io/badger/issues/598) (specifically doing key-only iteration as opposed to de-serializing each individual transaction result, just to delete the key)